### PR TITLE
AzureDevops | Add Seamless Proxy Ability (AST-67328)

### PR DESCRIPTION
**Pull request by AlvoBen**

### CODE REVIEW

**Comment by AlvoBen** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [a39e0b67\-7734\-422c\-9938\-1d35b8a01808](https://eu.ast.checkmarx.net/projects/0ada7c30-8fbf-4842-903d-a956ba0409ba/scans?id=a39e0b67-7734-422c-9938-1d35b8a01808&branch=feature%2Fbenalvo%2Fadd-proxy-support)
### Policy Management Violations \- Pull/merge request blocked
Policy Name|Rule(s)|Break Build
---|---|---
Phoenix\-Policy|New vulnerabilities of High, Medium and Low severity levels detected|true
### New Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
LOW|[Cxda14f253-4e52](https://devhub.checkmarx.com/cve-details/Cxda14f253-4e52)|Npm\-bluebird\-3\.7\.2|[Vulnerable Package](https://eu.ast.checkmarx.net/results/0ada7c30-8fbf-4842-903d-a956ba0409ba/a39e0b67-7734-422c-9938-1d35b8a01808/sca?internalPath=%2Fpackages%2FNpm-bluebird-3.7.2%2FpackageDetailsGql)
### Fixed Issues
Severity|Issue|Source File / Package
---|---|---
HIGH|[CVE-2022-25883](https://devhub.checkmarx.com/cve-details/CVE-2022-25883)|Npm\-semver\-6\.3\.0
HIGH|[Cxc7705965-e0f0](https://devhub.checkmarx.com/cve-details/Cxc7705965-e0f0)|Npm\-@babel/core\-7\.15\.0

File: src/main/client/HttpClient.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private readonly axiosConfig: AxiosRequestConfig;
+
+    constructor() {
+        this.axiosConfig = {
+            responseType: 'stream',
+            proxy: this.getProxyConfig(),
+        };
+    }
+
+    public getProxyConfig() {
```
Comment (miryamfoiferCX) [inline src/main/client/HttpClient.ts:17]: maybe better to save this value to avoid parsing the URL every time it’s called

Comment (AlvoBen) [reply src/main/client/HttpClient.ts:17]: it's being call once every httpClient object creation

**Review by OrShamirCM — APPROVED** [review]
Looks good!

File: src/main/client/Client.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import {AxiosResponse} from "axios";
+
+export interface Client {
```
Comment (OrShamirCM) [inline src/main/client/Client.ts:3]: Make sure the naming convention in TS is without "I" for interfaces. I believe it needs to be IClient ( that says that it's an interface)

**Pull request merged**